### ISSUE TITLE
ATI Rage 128 Pro for 86Box

### ISSUE THREAD

**Issue by JimboMumbo** [86Box/86Box#2343]
Besides 3dfx voodoo stuff, someone should add these two iconic graphic cards because they are nostalgic as it should be.

If you didn't like my idea, that's okay.

**Comment by Houfino** [comment]
Yes and we want...I want windows 98 to support a graphics card up to 256mb ..... Why didn't they? Thank you :-)

**Comment by TC1995** [comment]
no proper manuals.

**Comment by Cacodemon345** [comment]
NVIDIA GeForce 256 is out of question as it was released in 1999 and also potentially too powerful to be comfortably emulated.

ATi Rage 128 Pro may be surmountable.

NVIDIA RIVA TNT may get support instead as Nouveau supports it with 3D support as well and it also was released in 1998.

However, we need someone understanding how drawing 3D triangles actually work, as well as depth testing, should we actually go for adding more 3D cards.

**Comment by a-hurst** [comment]
> ATi Rage 128 Pro may be surmountable.

There's already been some progress on this front with the old Mac emulator DingusPPC, you can take a look at the source here!

https://github.com/dingusdev/dingusppc/blob/master/devices/video/atirage.cpp

QEMU also has some [preliminary work merged upstream](https://github.com/qemu/qemu/blob/master/hw/display/ati.c). If you can get a hold of the developers they've probably tracked down a lot of the relevant docs down, and might be interested in a collaboration!

**Comment by OBattler** [comment]
> NVIDIA GeForce 256 is out of question as it was released in 1999 and also potentially too powerful to be comfortably emulated.
> 
> ATi Rage 128 Pro may be surmountable.
> 
> NVIDIA RIVA TNT may get support instead as Nouveau supports it with 3D support as well and it also was released in 1998.
> 
> However, we need someone understanding how drawing 3D triangles actually work, as well as depth testing, should we actually go for adding more 3D cards.

Even if we were able to, by some magical miracle, get the Riva's implemented, we'd hit the problem that our current renderer framework is already too slow for the Voodoo 3 and Banshee, yet alone for the even newer Riva's.

So I'd say, nope, the Rivas are never going to happen.

But yes, I can see ATi Rage 128 Pro happen some day, as well as Cirrus Logic Laguna 3D and SGI IrisVision.

**Comment by JimboMumbo** [comment]
> But yes, I can see ATi Rage 128 Pro happen some day, as well as Cirrus Logic Laguna 3D and SGI IrisVision.

If I remember correctly, ATI Rage 128 Pro was supposedly to get windows 98 (or 95) working like a charm, so I don't have to worry about corrupted textures before (unless I will get better Frames per second as usual).

And as for Cirrus Logic Laguna 3D? Well, the other games seems fine however but when someone tries Half-Life, Incoming with 2MB and Thief Gold, they could see white textures. No wonder why Cirrus Logic abandons pc graphics like that :-\

**Comment by Raulshc** [comment]
#677 (**ATI 3D Rage Pro emulation**) is related with this issue. Some HW docs can be found in [http://www.vgamuseum.info](http://www.vgamuseum.info), as Programer's guide, Register Reference Guide and specifications.

**Comment by JimboMumbo** [comment]
> #677 (**ATI 3D Rage Pro emulation**) is related with this issue. Some HW docs can be found in http://www.vgamuseum.info, as Programer's guide, Register Reference Guide and specifications.

It should be easier than usual. It has a lot of ATI graphic stuff: https://archive.org/details/ATIDriverCollection20012005

**Comment by JimboMumbo** [comment]
Oh I forgot to say that someone made a comparison video of this: https://www.youtube.com/watch?v=aY0Nu-V0yQQ

(watch closely at 0:44 as you could see some distorted textures in Ignition, and 5:14 showing slow fps of Half Life)

**Comment by JimboMumbo** [comment]
[rrg-g04500-c_rev1.01_rage_128_pro_register_reference_guide_jan00 (1).pdf](https://github.com/86Box/86Box/files/8936288/rrg-g04500-c_rev1.01_rage_128_pro_register_reference_guide_jan00.1.pdf)
Well. Here goes ATI Rage 128 Pro Datasheet document.

**Comment by headsetguy37** [comment]
> [rrg-g04500-c_rev1.01_rage_128_pro_register_reference_guide_jan00 (1).pdf](https://github.com/86Box/86Box/files/8936288/rrg-g04500-c_rev1.01_rage_128_pro_register_reference_guide_jan00.1.pdf) Well. Here goes ATI Rage 128 Pro Datasheet document.

Thank you so much! That will be a huge help, I think.

**Comment by Raulshc** [comment]
Documents available:

- Programmer guide: [http://old.vgamuseum.info/images/stories/doc/ati/prg-215r3-00-10_rage_pro_and_derivatives_programmers_guide_mar00.pdf](http://old.vgamuseum.info/images/stories/doc/ati/prg-215r3-00-10_rage_pro_and_derivatives_programmers_guide_mar00.pdf)

- Register Reference guide: [http://old.vgamuseum.info/images/stories/doc/ati/rrg-g04500-c_rev1.01_rage_128_pro_register_reference_guide_jan00.pdf](http://old.vgamuseum.info/images/stories/doc/ati/rrg-g04500-c_rev1.01_rage_128_pro_register_reference_guide_jan00.pdf)

- Specification guide: [http://vgamuseum.info/images/doc/ati/rage128/gcs-c04100_rage_128_vr_and_128_gl_graphics_controller_specifications_nov98.pdf](http://vgamuseum.info/images/doc/ati/rage128/gcs-c04100_rage_128_vr_and_128_gl_graphics_controller_specifications_nov98.pdf)

**Comment by JimboMumbo** [comment]
I forgot the bios rom of ATI Rage 128 Pro because I had no clue what to do next
[rage128pro.zip](https://github.com/86Box/86Box/files/8987340/rage128pro.zip)
.

**Comment by JimboMumbo** [comment]
I'm not sure if I close my issue or leave it open, because I haven't say it a month ago...

**Comment by JimboMumbo** [comment]
I found out that ATI Rage 128 applies with a different ATI cards like All-In-Wonder 128, Xpert and Rage Fury. I suppose

**Comment by JimboMumbo** [comment]
Well.. time for me to close the issue since I haven't talked here.
